src/components/modals/AddUserModal.js

```diff
@@ -0,0 +1,818 @@
+import React, { Component, Fragment } from 'react';
+import { div, form, input, label, hh, h, select, option } from 'react-hyperscript-helpers';
+import { BaseModal } from '../BaseModal';
+import { User } from "../../libs/ajax";
+import { Alert } from '../Alert';
+import { USER_ROLES_UPPER } from '../../libs/utils';
+
+const ADMIN_ROLE_ID = 4;
+
+export const AddUserModal = hh(class AddUserModal extends Component {
```

Comment:
I think this class has too much embedded. I would suggest breaking out a lot of this into more granular components.